src/core/src/pass/serialize.cpp

```diff
@@ -885,8 +885,35 @@ class PaddingsFixer {
     }
 };
 
+bool is_correct_tag_name(const std::string& name) {
+    if (name.length() == 0) {
+        return false;
+    }
+    if (!std::all_of(name.begin(), name.end(), [](const int c) {
+            return std::isalnum(c) || (c == '_') || (c == '-') || (c == '.');
+        })) {
+        return false;
+    }
```

Comment:
```suggestion
    if (!std::all_of(name.begin(), name.end(), [](const int c) {
            return std::isalnum(c) || (c == '_') || (c == '-') || (c == '.');
        })) {
        return false;
    }
```
For empty string `std::all_of` should return false also.

Reply:
You are right, I just prefer explicit check :) I think I'll move it near name[0] access, just to prevent issues in case of reordering in the future. If this build fails - I'll apply, if ok - change as a separate pr

Reply:
No, `std::all_of` for an empty sequence returns `true`.